my-app/src/app/core/ui/header/header.component.ts

```diff
@@ -1,22 +1,25 @@
-import {Component, OnInit} from '@angular/core';
-import {AuthService} from '../../services/auth.service';
-import {User} from '../../../shared/models/user.model';
-import {Observable} from 'rxjs';
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '../../services/auth.service';
+import { User } from '../../../shared/models/user.model';
+import { Observable } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { AppState } from '../../../store/app.reducer';
+import { LoadUser } from '../../../store/auth/auth.actions';
 
 @Component({
   selector: 'cp-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss']
 })
 export class HeaderComponent implements OnInit {
-  public user$!: Observable<User>;
+  public user$!: Observable<User | undefined>;
```

Comment:
`user$ = this.store.select('auth', 'user');`

even better to create a separate selector for it to avoid dealing with string keys here. Components shouldn't be aware of state object structure.